ibis/backends/pandas/convert.py

```diff
@@ -83,3 +83,11 @@ def convert_String(cls, s, dtype, pandas_type):
         # TODO(kszucs): should switch to the new pandas string type and convert
         # object columns using s.convert_dtypes() method
         return s.map(str, na_action="ignore").astype(object)
+
+    @classmethod
+    def convert_Struct(cls, s, dtype, pandas_type):
+        # dt.normalize returns a frozendict, which doesn't play well
+        # with the pandas executor
+        # https://github.com/ibis-project/ibis/issues/8687
+        func = lambda x: x if x is pd.NA else dict(dt.normalize(dtype, x))
```

Comment:
I _think_ you'll need to recurse here for nested structs.

Reply:
dt.normalize will recurse, but that's not quite enough, that would lead to dict(frozendict(frozendict(..))). I assume we want dict all the way down. This seems hard to do.

better if we can address the linked bug, then just frozendict s all the way down and we don't need this workaround

Reply:
This has been fixed in [#Frozen](https://github.com/ibis-project/ibis/pull/8693) so I removed this workaround